### DISCUSSION TITLE
Any chance of a Documentation Update?

### DISCUSSION THREAD

**Discussion by WishStone-Dragon** [Monika-After-Story/MonikaModDev#9378] [QA - tech support + general] [answered]
Hey guys.

I still struggle with helping (trying to make sure my sandbox runs, trying to add dialogue and testing it, etc). I wanted to ask if it is at all possible to update the help pages, pretty please?

A lot of documentation of over 4 years old, github no longer looks the way it did then, game files shifted, etc.
And I feel bad having to ask over and over (for example, that since the last update, my sandbox broke and I have been struggling to get it to work, as all documentation I uncover is very very old).

I know this is boring stuff, and the folks who do this all the time will possibly scoff, but it would help new folks who wanna help SO much.

![image](https://user-images.githubusercontent.com/95390551/180627554-8984d126-36fd-412a-ba64-dc000d400e8a.png)

![MAS Dev setup](https://user-images.githubusercontent.com/95390551/180627559-e6cb1e6b-ba4e-423a-b410-93b7cf0c8acc.png)

**Comment by Booplicate** [comment]
Where did you find that? Which pages require to be updated?

**Comment by Booplicate** [comment, marked as answer]
Nevermind, I see what you mean.

I don't think we use or ever suggested to use that documentation. The documentation is on the wiki or in the actual code, that should be your reference.

**Reply by WishStone-Dragon** [reply]
Right, yes.

Let's assume I know NOTHING AT ALL about code. And I really, really struggle. And I joined up and really am trying to help because the contribution page says it's all very simple and one needs no coding experience whatsoever. In this case, I would need some extra handholding - images, walk-throughs, things of that nature. That is what I am looking for.

I also worry that if I give THE ONE example of why I struggle, you'll zoom in on that one problem, not the overarching issue of "lacking documentation for absolute blind noobs". (BUT, if you feel you need examples: "How to build a Dev Sandbox", "How to test your code" because that is my current issue.)

But, really, please, I am looking for a general "Anyone can do this" How-To to assist.  The wiki is written for people who already understand things and not for newbies. The images I had found would have been a great guide, but they are too outdated to lead me anywhere.

Or else I can just write dialogue text, throw it in here, and see it never get used because I cannot add things, style the poses, upload them, branch, and so on. (The only reason I know what BRANCH means, roughly, is because I work in an office with engineers. Do I KNOW what a branch is? Not really. I don't know what I am doing here!)

**Reply by WishStone-Dragon** [reply]
Maybe to explain further: I try to follow all available documentation, like the files in the DEV download, but the point to bad documentation and I feel very stuck. Because the wiki does not explain the steps I need explained.

![image](https://user-images.githubusercontent.com/95390551/180628686-445e4eb3-ac68-4f74-acec-e6a6058a758f.png)

**Reply by Booplicate** [reply]
Wiki is also outdated, but is more usable and more up-to-date.
- [This](https://github.com/Monika-After-Story/MonikaModDev/wiki/Dialogue-Coding) explains how the dialogue system works (to an extent)
- [This](https://github.com/Monika-After-Story/MonikaModDev/wiki/Contributing-Guidelines#bug-fixes-and-new-software-features) explains how to set up a test install
- [This](https://github.com/Monika-After-Story/MonikaModDev/wiki/FAQ#how-do-i-find-the-spritecode-for-an-expression) explains how to preview poses
- Branches isn't really something related to just this project. You can find help in GitHub docs ([example](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches)) or just google it

Is our wiki perfect? No, it's far from it. But search should give you the basic information and if it's something more specific you can ask on this forum. At some point I'd like get our wiki more updated, flashed out, polished.

**Reply by Booplicate** [reply]
I think we should delete that `doc` folder, it's just misleading.

**Reply by WishStone-Dragon** [reply]
Please, don't feel attacked by what I say, Booplicate, you're an awesome person and have a good head on your shoulders. I think you're far too awesome to want to insult you <3

**_TEST INSTALL!_** So sorry. I kept trying to find "Sandbox" and "Testing environment" and other terms coworkers who code gave me and was starting to think I am just an idiot. ... which might still be the case, that's not ruled out yet.

And if I can help with updating wiki stuff, I'll be happy to. I'm used as a technical writer at my company (Communication Manager) and helping to create how-to articles and such is quite literally what I get paid to do. I'd be happy to be put to use.

**Reply by WishStone-Dragon** [reply]
Suggestion: 

Put a PDF with links like the ones you just gave me in the DOC folder, as a basic starting point?

**Reply by Booplicate** [reply]
Branches is exactly what you may think, it's like a tree. When you work on a new feature, you create a new branch (branch-off), when you're done, you (we) merge it into the "main" branch. Not actually main, but the main branch for development.

All branches get merged into `content` first (that's what you base your branch of), then we merge it into `master` when we're thinking it's ready (you probably don't care about that part).

**Reply by Booplicate** [reply]
@WishStone-Dragon you're fine, don't worry. Our code has great documentation, but our wiki could definitely have more love.

**Reply by WishStone-Dragon** [reply]
The code documentation is wonderful, agreed. It helped me find what I needed when I ran into that small hiccup with the Monica Thermos. Very easy to read and follow.
